command/agent/config.go

```diff
@@ -606,6 +606,11 @@ type ServerConfig struct {
 	// for the EventBufferSize is 1.
 	EventBufferSize *int `hcl:"event_buffer_size"`
 
+	// JobMaxSourceSize limits the maximum size of a jobs source hcl/json
+	// before being discarded automatically. If unset, the maximum size defaults
+	// to 1 MiB. If the value is zero, no job sources will be stored.
```

Comment:
We're setting it to `1M` which is 1 million bytes, not 1024^2 bytes (https://go.dev/play/p/_F5kNg8Mtqn):

```suggestion
	// to 1 MB. If the value is zero, no job sources will be stored.
```

That being said, we set almost all our other byte size configs in powers-of-2 bytes and not powers-of-10 bytes, so I wouldn't be opposed to setting the default to `1 MiB` either.